magma/syntax/inline_combinational.py

```diff
@@ -44,6 +49,12 @@ def visit_Assign(self, node):
             node.targets[i] = ast.Name(self.assign_map[key], ast.Store())
         return node
 
+    def visit_Call(self, node):
+        if node.func.value.id in self.magma_ids and node.func.attr == "wire":
```

Comment:
Hmm, rather than using this "magma_ids" pattern, how about just passing the "env" to this visitor and calling "eval" on node.func inside the env? This should support more general patterns and we're passing around/evaling inside the env anyways.

Reply:
You could use "astor.to_source" to dump the AST to a string that you can eval with env.

Reply:
sure, seems `eval(node.func.value.id, {}, self.env)` is working